File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l, h = 0, len(s) - 1
        
        while l < h:
            if s[l].isalpha() == False and s[l].isdigit() == False:
                l += 1
            elif s[h].isalpha() == False and s[h].isdigit() == False:
                h -= 1
            elif s[l].lower() != s[h].lower():
                return False
            else:
                l += 1
                h -= 1
        
        return True

# Example usage
def main():
    s = "as&78&2332&& sadha 77a8s&8^ 233287sa";

    obj = Solution()
    print(obj.isPalindrome(s))

if __name__ == "__main__":
    main()

File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/valid-palindrome.cpp
#include <iostream>
#include <string>

class Solution {
public:
    bool isPalindrome(std::string s) {
        int l = 0, h = s.size() - 1;
        
        while (l < h) {
            if (!isalpha(s[l]) and !isdigit(s[l])) l++; // You can use isalnum() function to check both at once
            else if (!isalpha(s[h]) and !isdigit(s[h])) h--;
            else if (tolower(s[l]) != tolower(s[h])) return false;
            else l++, h--;
        }
        
        return true;
    }
};

// Example usage
int main() {
    Solution solution;
    std::string s = "I am the king ???? of the world! dlrow eht fo !!!! gnik eht ma I";

    bool result = solution.isPalindrome(s);
    std::cout << (result ? "true" : "false") << std::endl;
}